EIPS/eip-5334.md

```diff
@@ -0,0 +1,113 @@
+
+## Specification
+
+The keywords "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY" and "OPTIONAL" in this document are to be interpreted as described in RFC 2119.
+
+### Contract Interface
+Solidity Interface with NatSpec & OpenZeppelin v4 Interfaces (also available at [`IERC5334.sol`](../assets/eip-5334/IERC5334.sol)):
+
+```solidity
+interface IERC5334 {
+
+    // Logged when the user of a NFT is changed or expires and level is changed
+    /// @notice Emitted when the `user` of an NFT or the `expires` of the `user` is changed or the user `level` is changed
+    /// The zero address for user indicates that there is no user address
+    event UpdateUser(uint256 indexed tokenId, address indexed user, uint64 expires, uint8 level);
+
+    /// @notice set the user and expires and level of a NFT
+    /// @dev The zero address indicates there is no user
+    /// Throws if `tokenId` is not valid NFT
+    /// @param user  The new user of the NFT
+    /// @param expires  UNIX timestamp, The new user could use the NFT before expires
+    /// @param level user level
+    function setUser(uint256 tokenId, address user, uint64 expires, uint8 level) external;
+
+    /// @notice Get the user address of an NFT
+    /// @dev The zero address indicates that there is no user or the user is expired
+    /// @param tokenId The NFT to get the user address for
+    /// @return The user address for this NFT
+    function userOf(uint256 tokenId) external view returns(address);
+
+    /// @notice Get the user expires of an NFT
+    /// @dev The zero value indicates that there is no user
+    /// @param tokenId The NFT to get the user expires for
+    /// @return The user expires for this NFT
+    function userExpires(uint256 tokenId) external view returns(uint256);
+
+    /// @notice Get the user level of an NFT
+    /// @dev The zero value indicates that there is no user
+    /// @param tokenId The NFT to get the user level for
+    /// @return The user level for this NFT
+    function userLevel(uint256 tokenId) external view returns(uint256);
+}
+```
+
+The `userOf(uint256 tokenId)` function MAY be implemented as `pure` or `view`.
+
+The `userExpires(uint256 tokenId)` function MAY be implemented as `pure` or `view`.
+
+The `userLevel(uint256 tokenId)` function MAY be implemented as `pure` or `view`.
+
+The `setUser(uint256 tokenId, address user, uint64 expires)` function MAY be implemented as `public` or `external`.
+
+The `UpdateUser` event MUST be emitted when a user address is changed or the user expires is changed or the user level is changed.
+
+The `supportsInterface` method MUST return `true` when called with `0xad092b5c`.
```

Comment:
`0xad092b5c` is the interfaceId of [ERC-4907](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-4907.md),  this value should be changed.